File: api/app/models/apiary.py
from enum import Enum as PyEnum
from typing import Optional

from beanie import Document

from .base import TimestampMixin


class ApiaryStatus(str, PyEnum):
    HEALTHY = "HEALTHY"
    WARNING = "WARNING"
    ALERT = "ALERT"


class Apiary(Document, TimestampMixin):
    id: str  # type: ignore[assignment]
    name: str
    location: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    status: ApiaryStatus = ApiaryStatus.HEALTHY

    class Settings:
        name = "apiaries"

File: composeApp/src/commonMain/kotlin/com/beekeeper/app/domain/model/Apiary.kt
package com.beekeeper.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Apiary(
    val id: String,
    val name: String,
    val location: String,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val hiveCount: Int,
    val status: ApiaryStatus
)

enum class ApiaryStatus {
    HEALTHY,
    WARNING,
    ALERT
}
